src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -383,12 +395,11 @@ def _filter_file(tarinfo):
 
         return source_upload_location.relative_path
     except Exception as err:
-        raise CLIError(err)
-    finally:
         if os.path.exists(tar_file_path):
             logger.debug(
                 "Starting to delete the archived source code from '{}'.".format(tar_file_path))
             os.remove(tar_file_path)
+        raise CLIError(err)
```

Comment:
What happens if `os.remove` failed?

Reply:
In the current context, I do not think it necessary to check if os.remove fails. It is true that os.remove triggers an OSError while the argument passed to it is a directory rather than a file. Actually the way we construct the "tar_file_path" confirms that it is file rather than a directory.
tar_file_path = os.path.join(tempfile.gettempdir(),
                                 "source_archive_{}.tar.gz".format(hash(os.times())))